101-hdinsight-linux-ssh-publickey/azuredeploy.json

```diff
@@ -99,10 +94,10 @@
         "storageProfile": {
           "storageaccounts": [
             {
-              "name": "[concat(parameters('clusterStorageAccountName'),'.blob.core.windows.net')]",
+              "name": "[replace(replace(concat(reference(concat('Microsoft.Storage/storageAccounts/', variables('storageAccountName')), variables('defaultApiVersion')).primaryEndpoints.blob),'https:',''),'/','')]",
```

Comment:
It should be sufficient to use variables('storageAccountName') here without prefixing it with 'Microsoft.Storage/storageAccounts/'